### DISCUSSION TITLE
OAuth2-Proxy support for authentication

### DISCUSSION THREAD

**Discussion by jared252016** [NginxProxyManager/nginx-proxy-manager#2799] [Ideas]
Add support for OAuth2-Proxy and proxy_auth as an authentication method, with API support.

This is extremely easy to do if doable per proxy-host, but setting up auth like instead of basic auth would be ideal.

**Comment by LeonardoIz** [comment]
For authentication it is better to use Authelia or Authentik, both work very well with npm

**Reply by phulstaert** [reply]
Most people won't change their current authentication system because they want to use npm.
Authentik uses a proxy provider that works with npm (although they have a [huge error](https://github.com/goauthentik/authentik/discussions/10004) in their config)

It would be wise to implement the OAuth2 authentication protocol for access to npm itself AND access to subdomains.
A good and easy implementation would 'lure' a lot of SSO users.

**Reply by phulstaert** [reply]
I also see there is a "openidc" branch with nice gui support. A real pitty this is not documented or integrated in the latest release. Now you have to choose between the latest updates or OpenID Connect Support

**Comment by Robert-D** [comment]
Yes, this would be a wonderful feature to integrate with our existing Identity Provider and allow only authenticated sessions to proxy through to hosts.
